src/stub/core_impl/thread_extra_info.rs

```diff
@@ -0,0 +1,40 @@
+use super::prelude::*;
+use crate::protocol::commands::ext::ThreadExtraInfo;
+use crate::protocol::SpecificIdKind;
+use crate::target::ext::base::BaseOps;
+
+impl<T: Target, C: Connection> GdbStubImpl<T, C> {
+    pub(crate) fn handle_thread_extra_info<'a>(
+        &mut self,
+        res: &mut ResponseWriter<'_, C>,
+        target: &mut T,
+        command: ThreadExtraInfo<'a>,
+    ) -> Result<HandlerStatus, Error<T::Error, C::Error>> {
+        let ops = match target.base_ops() {
+            BaseOps::SingleThread(_) => return Ok(HandlerStatus::Handled),
+            BaseOps::MultiThread(ops) => match ops.support_thread_extra_info() {
+                Some(ops) => ops,
+                None => return Ok(HandlerStatus::Handled),
+            },
+        };
+
+        crate::__dead_code_marker!("thread_extra_info", "impl");
+
+        let handler_status = match command {
+            ThreadExtraInfo::qThreadExtraInfo(info) => {
+                if let SpecificIdKind::WithId(tid) = info.id.tid {
+                    let size = ops
+                        .thread_extra_info(tid, info.buf)
+                        .map_err(Error::TargetError)?;
+                    let data = info.buf.get(..size).ok_or(Error::PacketBufferOverflow)?;
+
+                    res.write_hex_buf(data)?;
+                }
```

Comment:
You'll want to add an extra else arm here:

```rust
} else {
    return Err(Error::PacketParse(
        crate::protocol::PacketParseError::MalformedCommand,
    ))
}
```

The _best_ solution would actually be to go the extra mile and introduce a new `ThreadId` variant at the `protocol` level that doesn't support the `All` variant. Something like `ConcreteThreadId` (not a great name, but whatever). That way, an invalid thread-id would be caught at the packet parse level, rather than within the handler.

It would be nice to implement that latter solution, but it's fine if you'd like to stick with the quick-and-dirty fix I proposed above.

Reply:
I like that - I added a `ConcreteThreadId` type.  Naming is hard - I couldn't come up with another alternative that I felt expressed things more clearly.

As suggested this simplified the handler logic.